File: polyglot/seis_native/haskell/SeisNativeKernel.hs
module SeisNativeKernel
  ( NativeRoadmapItem(..)
  , roadmap
  , topLane
  ) where

data NativeRoadmapItem = NativeRoadmapItem
  { lane :: String
  , score :: Int
  , language :: String
  } deriving (Eq, Show)

roadmap :: [NativeRoadmapItem]
roadmap =
  [ NativeRoadmapItem "Apple First" 100 "Swift"
  , NativeRoadmapItem "Data AI" 88 "Python"
  , NativeRoadmapItem "Systems" 84 "Rust"
  , NativeRoadmapItem "Android" 76 "Kotlin"
  , NativeRoadmapItem "Windows" 72 "CSharp"
  , NativeRoadmapItem "Infrastructure" 70 "Go"
  ]

topLane :: NativeRoadmapItem
topLane = head roadmap

File: polyglot/seis_native/ocaml/seis_native_kernel.ml
type native_roadmap_item = {
  lane : string;
  score : int;
  language : string;
}

let roadmap = [
  { lane = "Apple First"; score = 100; language = "Swift" };
  { lane = "Data AI"; score = 88; language = "Python" };
  { lane = "Systems"; score = 84; language = "Rust" };
  { lane = "Android"; score = 76; language = "Kotlin" };
  { lane = "Windows"; score = 72; language = "CSharp" };
  { lane = "Infrastructure"; score = 70; language = "Go" };
]

let top_lane = List.hd roadmap
